x-pack/legacy/plugins/monitoring/server/lib/setup/collection/get_collection_status.js

```diff
@@ -223,6 +237,18 @@ function isBeatFromAPM(bucket) {
   return get(beatType, 'buckets[0].key') === 'apm-server';
 }
 
+async function hasNecessaryPermissions(req) {
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('data');
+  const response = await callWithRequest(req, 'transport.request', {
+    method: 'POST',
+    path: '/_security/user/_has_privileges',
+    body: {
+      cluster: ['monitor'],
+    }
+  });
+  return get(response, 'has_all_requested', true);
```

Comment:
Shouldn't this default to `false` instead of `true`?

Reply:
Good point. I'm actually not sure, but you're probably right. I borrowed it [from this code](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/monitoring/server/lib/elasticsearch/verify_monitoring_auth.js#L55)

Reply:
Fixed in https://github.com/elastic/kibana/pull/50421/commits/e1be90f5e9279bca1f6216681e8792c1a87b8a09

Reply:
Oh, I think this is incase the await gets rejected? Shouldn't we do something like this then:
`return response && response.has_all_requested`

I'm probably missing something obvious here

Reply:
Eh, sorry. Didn't see your comments before I added the one above this one